### ISSUE TITLE
Performance enhancement request: Faster approval

### ISSUE THREAD

**Issue by ToyB0x** [GoogleCloudPlatform/jit-access#221]
Hi, Thank you for creating such a useful tool.

If I apply for only one role, the approval process will be completed in a few seconds,
but it seems that the process takes a few seconds longer each time the role increases.
(For example, if I apply for 10 roles, it will take approximately 1 minute to complete the process)

If this application processes internally serially, there seems to be room for speeding it up by using parallel processing.

(I confirmed the above behavior in self-approval mode of v1.4.1 and v1.5.0)

**Comment by jpassing** [comment]
You're right, activating multiple roles can be quite slow and the reason is that the roles are being processed sequentially. There's definitely room for improvement here.

**Comment by jpassing** [comment]
Previous versions performed one `analyzeIamPolicy` API request and one IAM `setIamPolicy` request per role. Obviously, that wasn't very efficient, particularly given that `analyzeIamPolicy` API request are expensive.

[Version 1.6](https://github.com/GoogleCloudPlatform/jit-access/releases/tag/1.6.0) improves that behavior and only performs a single  `analyzeIamPoliocy`/`effectiveIamPolicies.batchGet` API request, regardless how may roles you're trying to activate. However, it still performs one IAM `setIamPolicy` request per role. 

There's still room for improvement, but the performance should be noticeably better now. I'm therefore inclined to close this issue for now.
